### PR TITLE
Clarify mask/dseg/probseg in common derivatives

### PR DESCRIPTION
**Pull request by effigies**
This PR does the following:

1) Remove conceptual overlap of binary masks and binary dsegs by standardizing on mask and updating the example to show that a single ROI should be renamed to `mask`. I left it under `dseg` to make the connection plain, as I don't think anything would be gained by moving it to mask (which has an ROI example already). This follows a brief discussion with @Lestropie and @edickie in https://github.com/bids-standard/bids-specification/pull/265#discussion_r432818644/https://github.com/bids-standard/bids-specification/pull/265#discussion_r432874774.
2) I added a preface to segmentations (removing parcellation from the section title) to place a conceptual hierarchy of type (discrete vs probabilistic) and a subordinate consideration of representation (volume vs surface) and further that we are not currently specifying probabilistic surface segmentations. This follows a discussion with @Lestropie and @satra in https://github.com/bids-standard/bids-specification/pull/265#discussion_r432816173. While there is a desire to specify here, I'm refraining from doing so. It can be added without breaking backwards compatibility in the future.
3) I also addressed @satra's https://github.com/bids-standard/bids-specification/pull/265#discussion_r432845797 while here.

Please read and let me know what you think.

### CODE REVIEW

File: src/05-derivatives/03-imaging.md [hunk 1]
```diff
@@ -189,24 +204,25 @@ pipeline/
             sub-001_space-orig_dseg.json
 ```
 
-A segmentation could be a binary mask that functions as a discrete `label` for a
-single structure. In this case, the label key must be used to specify the
-corresponding structure. For example:
+A segmentation can be used to generate a binary mask that functions as a
+discrete "label" for a single structure.
+In this case, the mask suffix MUST be used, and the `desc` entity SHOULD be used
```
[the remark this answers is not in the document]
Comment (edickie) [reply src/05-derivatives/03-imaging.md:209]: Good point. I can see where label and desc might want to be together.

File: src/05-derivatives/03-imaging.md [hunk 2]
```diff
@@ -153,9 +153,24 @@ manual_masks/
             sub-001_desc-tumor_mask.json
 ```
 
-## Segmentations and parcellations
+## Segmentations
 
-Common JSON metadata fields:
+A *segmentation* is a labeling of a spatial image with anatomical structures
```
[the remark this answers is not in the document]
Comment (effigies) [reply src/05-derivatives/03-imaging.md:158]: I find this suggestion really hard to parse.

Are there examples of 1D signals that will be segmented in neuroimaging that really motivates this? I can try to offer some wording that would cover those cases.

File: src/05-derivatives/03-imaging.md [hunk 3]
```diff
@@ -153,9 +153,24 @@ manual_masks/
...
+such that each voxel is identified with a label or a combination of labels.
+A *discrete segmentation* represents each tissue class with a unique integer
```
[the remark this answers is not in the document]
Comment (effigies) [reply src/05-derivatives/03-imaging.md:160]: Providing multiple equivalent ways of representing identical data is not really a goal of BIDS, and using the same suffix to enable multiple encodings is getting into territory that's hard to validate and hard to use downstream. If somebody very strongly wants to expand a label file out into ROI volumes, they can use multiple masks or call it a probseg that's all 0s and 1s.

**Comment by effigies** [comment]
@edickie @oesteban Thanks for the feedback! I applied your suggestions RE labels for masks, and tried to move from "tissue class" to "anatomical structure". Let me know what you think.

File: src/05-derivatives/03-imaging.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lestropie) [reply src/05-derivatives/03-imaging.md:161]: Epilepsy EEG markup where there are multiple different seizure types? This is coming from no familiarity with EEG BIDS whatsoever; it just came to mind. 

Another might be a [discrete fixel-based parcellation](https://mrtrix.readthedocs.io/en/latest/getting_started/image_data.html#fixel-image-directory-format); but whether such data are 1D or 4D is a matter of perspective :upside_down_face:

File: src/05-derivatives/03-imaging.md [hunk 4]
```diff
@@ -153,9 +156,25 @@ manual_masks/
...
 
-Common JSON metadata fields:
+A *segmentation* is a labeling of a spatial image with anatomical structures
+such that each location (for example, a voxel or a surface vertex) is identified
+with a label or a combination of labels.
+A *discrete segmentation* represents each structure with a unique integer
+label.
+A *probabilistic segmentation* represents each structure as values between
+0 and 1 (inclusive) at each location in the image, and one volume/frame per
+structure may be concatenated in a single file.
```
Comment (Lestropie) [inline src/05-derivatives/03-imaging.md:168]: Sorry, I'm gonna poke at this again :grimacing: 

This is technically another location where there are two different concepts getting mushed together (yes, that is the technical term):

- How are multiple segmented structures represented in a single file?
    - Within a single volume
    - Across volumes
- What data type is used to represent the segmentation?
    - Binary
    - Floating-point

For continuous / "probabilistic" data, storing multiple structures in a single volume is impossible, and so necessitates storing them across volumes.

For discrete data, we're accustomed to such being represented in a single volume / frame, using unique integer labels. This however enforces a one-to-one mapping between "elements" (e.g. voxels, vertices) and labels. In some instances (TractSeg comes to mind, as well as internal developments here), it's possible for one voxel to be included in multiple binary segmentations, at which point using unique integer labels does not work and one would need a 4D image. 

(In a way this could be thought of as a 4D `_mask.nii[.gz]` image as opposed to / in addition to a `_dseg` image...)

I don't actually have a fixed opinion on what to do here (if anything); just seeing if the spec can be made to cover the fundamentals, rather than unnecessarily excluding anything that strays slightly from the current most common use cases.

File: src/05-derivatives/03-imaging.md [hunk 5]
```diff
@@ -166,9 +185,10 @@ Common JSON metadata fields:
 
 ### Discrete Segmentations
 
-Discrete segmentations of brain tissue represent each tissue class with a unique
-integer label in a 3D volume. See [Anatomical Labels](#anatomical-labels) for interpretation how
-integer values map to tissue classes.
+Discrete segmentations of brain tissue represent multiple anatomical structures
+(such as tissue class or Brodmann area) with a unique integer label in a 3D volume.
+See [Anatomical Labels](#anatomical-labels) for interpretation how integer values
+map to anatomical structures.
```
Comment (Lestropie) [inline src/05-derivatives/03-imaging.md:191]:
```suggestion
See [Anatomical Labels](#anatomical-labels) for interpretation regarding how
integer values map to anatomical structures.
```

File: src/05-derivatives/03-imaging.md [hunk 6]
```diff
@@ -189,25 +209,27 @@ pipeline/
...
-See [Anatomical Labels](#anatomical-labels) for reserved key values for `label`.
-
 ### Probabilistic Segmentations
 
-Probabilistic segmentations of brain tissue represent a single tissue class with
-values ranging from 0 to 1 in individual 3D volumes or across multiple frames.
-Similarly to a discrete, binary segmentation, the label key can be used to
-specify the corresponding structure.
+Probabilistic segmentations of brain tissue represent a single anatomical
+structure with values ranging from 0 to 1 in individual 3D volumes or across
```
Comment (Lestropie) [inline src/05-derivatives/03-imaging.md:229]:
```suggestion
structure with values in the range `[0.0,1.0]` in individual 3D volumes or across
```

(trying to indicate the the indicated range is inclusive of the bounding values)
